spec/requests/contributions_spec.rb

```diff
@@ -18,10 +18,47 @@
   describe "GET /index" do
     it "renders a successful response" do
       create(:listing)
-      get contributions_public_url
-      skip # TODO - get this working instead of listingscontroller
+      get contributions_url
       expect(response).to be_successful
     end
-  end
 
+    it 'allows asking for a specific subtype of listing' do
```

Comment:
The tests here are copied/pasted from requests/listings_spec.rb. I'm fine with removing the duplication whenever you're ready